src/pages/AppList/AppListComponent.tsx

```diff
@@ -34,17 +34,13 @@ type AppListComponentState = {
 };
 
 type AppListComponentProps = {
-  onError: PropTypes.func;
   pagination: Pagination;
-  queryParam: PropTypes.func;
-  onParamChange: PropTypes.func;
+  pageHooks: ListPage.Component<any, any>;
```

Comment:
Can a type be added here ?

Reply:
@lucasponce right, it's a shortcut I did, to avoid having an interface that declares all the functions. My take is that it's acceptable here because this interface would else redeclare pretty much all ListPage.Component functions, so it wouldn't bring a lot of value but would add boilerplate code.

Let me commit something like this, and we'll see if it's better or not.

Reply:
For me is fine as a temporal scenario, just understanding that KIALI-1477 should refactor this.
But this "pageHooks" makes code more difficult to read as I need to check which is declared on the parent.

Reply:
See the last commit, using an interface, it makes the things clearer.

> But this "pageHooks" makes code more difficult to read as I need to check which is declared on the parent.

Maybe there's this little drawback while reading; but on another hand, while writing I find it better, thanks to the IDE autocompletion which makes it easy to see the available hooks from a single property. But I'm also ok if you prefer to rollback to individual hook properties, no strong opinion here.

Reply:
Last commit is great.
Adding an interface it clarifies much better what is expected.
Thanks for adding it.